jupyter_notebook/notebookapp.py

```diff
@@ -261,6 +263,22 @@ def init_handlers(self, settings):
         new_handlers.append((r'(.*)', Template404))
         return new_handlers
 
+    def _create_widget_shim_handlers(self, settings):
+        """Creates a handler that shims the loading of widgets
+        JS files for the old /static/widgets/js/*.js path."""
+        class DeprecationHandler(IPythonHandler):
+            def get(self, url_path):
+                self.set_header("Content-Type", 'text/javascript')
+                self.finish("""
+                    console.warn('`/static/widgets/js` is deprecated.  Use `/nbextensions/widgets/widgets/js` instead.');
+                    define(['%s'], function(x) { console.log('...loaded...'); return x; });
+                """ % url_path_join(settings['base_url'], 'nbextensions', 'widgets', 'widgets', url_path))
```

Comment:
@minrk this is driving me crazy- for the case of http://localhost:8888/deprecatedwidgets/js/init.js  this yields:

```js
console.warn('`/static/widgets/js` is deprecated.  Use `/nbextensions/widgets/widgets/js` instead.');
define(['/nbextensions/widgets/widgets/js/init.js'], function(x) { console.log('...loaded...'); return x; });
```

In the notebook, when I require this file, the warning shows in the console.  After the ~5sec require js timeout, a load timeout error is shown, no 404s in the network panel of the browser debugging tools, and the "...loaded..." text is never printed.  HOWEVER, when I try to access the file manually by navigating to `http://localhost:8888/nbextensions/widgets/widgets/js/init.js` it loads just fine!